From 2ace6ff95838a2614d1129e8386e7790582ee517 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Fri, 8 Sep 2017 09:28:19 -0400
Subject: [PATCH 56/80] kmssink: Don't assume buffer pool won't touch buf
 pointer

Just for extra safety, let's not assume that gst_buffer_pool_acquire()
won't ever touch the buf pointer.
---
 sys/kms/gstkmssink.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index 2c6cc7f6a..da0aee804 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -1187,9 +1187,8 @@ gst_kms_sink_copy_to_dumb_buffer (GstKMSSink * self, GstBuffer * inbuf)
   GstFlowReturn ret;
   GstVideoFrame inframe, outframe;
   gboolean success;
-  GstBuffer *buf;
+  GstBuffer *buf = NULL;
 
-  buf = NULL;
   if (!gst_buffer_pool_set_active (self->pool, TRUE))
     goto activate_pool_failed;
 
@@ -1223,13 +1222,13 @@ activate_pool_failed:
   {
     GST_ELEMENT_ERROR (self, STREAM, FAILED, ("failed to activate buffer pool"),
         ("failed to activate buffer pool"));
-    goto bail;
+    return NULL;
   }
 create_buffer_failed:
   {
     GST_ELEMENT_ERROR (self, STREAM, FAILED, ("allocation failed"),
         ("failed to create buffer"));
-    goto bail;
+    return NULL;
   }
 error_copy_buffer:
   {
-- 
2.11.0

